starter-code/routes/drones.js

```diff
@@ -1,21 +1,40 @@
 const express = require('express');
-
+const Drone = require('../models/drones')
 // require the Drone model here
 
 const router = express.Router();
 
 
 router.get('/drones', (req, res, next) => {
-  // Iteration #2
+  Drone.find({}, (err, drones) => {
+    if (err) {
+      next(err);
+    } else {
+      res.render('drones', {drones})
+    }
+  })
 });
 
-
 router.get('/drones/new', (req, res, next) => {
-  // Iteration #3
+  res.render('drones/new')
 });
 
+
 router.post('/drones', (req, res, next) => {
-  // Iteration #3
+  const newDroneInfo = {
+    droneName: req.body.droneName,
+    propellers: req.body.propellers,
+    maxSpeed: req.body.maxSpeed
+  }
+  const newDrone = new Drone(newDroneInfo);
+
+  newDrone.save((err) => {
+    if (err) {
+      next(err);
+    } else {
+      return res.redirect('../drones')
```

Comment:
You don't need the return neither  ../, is enough with res.redirect('/drones');
And don't forget the ;